### PR TITLE
Upgraded Ember Cli and Ember + Patch for sorting and footer

### PR DESCRIPTION
**Pull request by dcheng168**
- Upgraded Ember cli to 2.4.3
- Upgraded Ember to 2.4.5
- Removed {{view}}
- fixed sorting
- fixed footer destory

### CODE REVIEW

File: addon/components/llama-footer/component.js [hunk 1]
```diff
@@ -18,6 +18,8 @@ var LlamaFooter = Em.Component.extend({
 	columngroups: null,
 	rows: null,
 
+	isConstructor: false,
```
Comment (j-) [inline addon/components/llama-footer/component.js]: I would prefer something like `didConstruct`.

File: addon/components/llama-footer/component.js [hunk 2]
```diff
@@ -58,7 +62,10 @@ var LlamaFooter = Em.Component.extend({
 	}),
 
 	willDestroy: function () {
-		this.get('data').destroy();
+		if (this.get('isConstructor')) {
+			let footerController = this.get('data');
```
Comment (j-) [inline addon/components/llama-footer/component.js]: Should be `const`.

File: addon/components/llama-header-cell/component.js [hunk 3]
```diff
@@ -40,9 +40,10 @@ var LlamaHeaderCell = LlamaCell.extend({
 
 	sortByThis: computed('sortProperties.firstObject', 'column.sortBy', {
 		get: function () {
-			var sortBy = this.get('sortProperties.firstObject');
-			var thisColumn = this.get('column.sortBy');
-			return sortBy === thisColumn;
+			const sortBy = this.get('sortProperties.firstObject');
+			const sortedColumn = sortBy.split(':', 1)[0];
```
Comment (j-) [inline addon/components/llama-header-cell/component.js]: Nice.

File: package.json [hunk 4]
```diff
@@ -32,29 +32,30 @@
...
+    "loader.js": "^4.0.1"
   },
   "dependencies": {
-    "ember-cli-babel": "^5.1.5",
-    "ember-cli-htmlbars": "^1.0.1",
+    "ember-cli-htmlbars": "^1.0.3",
     "ember-cli-htmlbars-inline-precompile": "^0.3.1",
-    "ember-legacy-controllers": "0.0.1",
-    "ember-legacy-views": "^0.2.0"
+    "ember-cli-babel": "^5.1.6"
```
Comment (j-) [inline package.json]: Can you make sure these deps are in alphabetical order (default for npm)?

**Comment by j-** [comment]
Good stuff :shipit:

File: addon/components/llama-footer/component.js (hunk 1 above)
Comment (dcheng168) [inline addon/components/llama-footer/component.js]: done

File: addon/components/llama-footer/component.js (hunk 2 above)
Comment (dcheng168) [inline addon/components/llama-footer/component.js]: done

**Pull request merged**